### PR TITLE
browser(firefox): introduce global proxy

### PR DESCRIPTION
**Pull request by aslushnikov**
This will be used instead of messing up user preferences for proxy
setup.

### CODE REVIEW

File: browser_patches/firefox/juggler/NetworkObserver.js [hunk 1]
```diff
@@ -571,7 +571,7 @@ class NetworkObserver {
       applyFilter: (channel, defaultProxyInfo, proxyFilter) => {
         const originAttributes = channel.loadInfo && channel.loadInfo.originAttributes;
         const browserContext = originAttributes ? this._targetRegistry.browserContextForUserContextId(originAttributes.userContextId) : null;
-        const proxy = browserContext ? browserContext.proxy : null;
+        const proxy = (browserContext ? browserContext.proxy : undefined) || this._targetRegistry.globalProxy();
```
Comment (yury-s) [inline browser_patches/firefox/juggler/NetworkObserver.js:574]: `(browserContext ? browserContext.proxy : undefined)` => `browserContext && browserContext.proxy` ?

File: browser_patches/firefox/juggler/protocol/BrowserHandler.js [hunk 2]
```diff
@@ -155,6 +155,10 @@ class BrowserHandler {
     this._targetRegistry.browserContextForId(browserContextId).proxy = { type, host, port, bypass };
   }
 
+  async setGlobalProxy({type, host, port, bypass}) {
```
Comment (yury-s) [inline browser_patches/firefox/juggler/protocol/BrowserHandler.js:158]: How about setBrowserProxy/setContextProxy?

**Review by yury-s — APPROVED** [review]

File: browser_patches/firefox/juggler/NetworkObserver.js (hunk 1 above)
Comment (aslushnikov) [reply browser_patches/firefox/juggler/NetworkObserver.js]: done.

File: browser_patches/firefox/juggler/protocol/BrowserHandler.js (hunk 2 above)
Comment (aslushnikov) [reply browser_patches/firefox/juggler/protocol/BrowserHandler.js]: done.

**Review by yury-s — APPROVED** [review]

**Pull request merged**